tests/test_bot.py

```diff
@@ -3196,3 +3202,38 @@ async def test_arbitrary_callback_data_get_chat_no_pinned_message(
         finally:
             bot.callback_data_cache.clear_callback_data()
             bot.callback_data_cache.clear_callback_queries()
+
+
+@pytest.mark.skipif(not TEST_WITH_OPT_DEPS, reason="Optional dependencies not installed")
+class TestBotWithOptDeps:
```

Comment:
Sorry to be pendantic here - but … The main part of the test suite should run with the optional dependencies as one run IMO, i.e. `pytest -v`. So far only the logic for the optional dependencies *not* being installed was extracted to special classes and is tested in the special call to pytest in the workflow.